enclave/src/polkadex.rs

```diff
@@ -174,62 +180,83 @@ impl PolkadexAccountsStorage {
     // Aquire lock on proxy_registry
     let mutex = load_proxy_registry()?;
-    let mut proxy_storage: SgxMutexGuard<PolkadexAccountsStorage> = mutex.lock().unwrap();
+    let mut proxy_storage: SgxMutexGuard<PolkadexAccountsStorage> = mutex
+        .lock()
+        .map_err(|_| AccountRegistryError::CouldNotGetMutex)?;
     Ok(proxy_storage.add_main_account(main_acc))
 }
 
-pub fn remove_main_account(main_acc: AccountId) -> Result<(), GatewayError> {
+pub fn remove_main_account(main_acc: AccountId) -> Result<(), AccountRegistryError> {
     // Aquire lock on proxy_registry
     let mutex = load_proxy_registry()?;
-    let mut proxy_storage: SgxMutexGuard<PolkadexAccountsStorage> = mutex.lock().unwrap();
+    let mut proxy_storage: SgxMutexGuard<PolkadexAccountsStorage> = mutex
+        .lock()
+        .map_err(|_| AccountRegistryError::CouldNotGetMutex)?;
     Ok(proxy_storage.remove_main_account(main_acc))
 }
 
-pub fn add_proxy(main_acc: AccountId, proxy: AccountId) -> Result<(), GatewayError> {
+pub fn add_proxy(main_acc: AccountId, proxy: AccountId) -> Result<(), AccountRegistryError> {
     // Aquire lock on proxy_registry
     let mutex = load_proxy_registry()?;
-    let mut proxy_storage: SgxMutexGuard<PolkadexAccountsStorage> = mutex.lock().unwrap();
+    let mut proxy_storage: SgxMutexGuard<PolkadexAccountsStorage> = mutex
+        .lock()
+        .map_err(|_| AccountRegistryError::CouldNotGetMutex)?;
     Ok(proxy_storage.add_proxy(main_acc, proxy))
 }
 
-pub fn remove_proxy(main_acc: AccountId, proxy: AccountId) -> Result<(), GatewayError> {
+// pub fn check_main_account(acc: AccountId) -> SgxResult<bool> {
+//     // Aquire lock on proxy_registry
+//     let mutex = load_proxy_registry()?;
+//     let mut proxy_storage: SgxMutexGuard<PolkadexAccountsStorage> = mutex.lock().unwrap();
+//     Ok(proxy_storage.accounts.contains_key(&acc.encode()))
+// }
+
+pub fn remove_proxy(main_acc: AccountId, proxy: AccountId) -> Result<(), AccountRegistryError> {
     // Aquire lock on proxy_registry
     let mutex = load_proxy_registry()?;
-    let mut proxy_storage: SgxMutexGuard<PolkadexAccountsStorage> = mutex.lock().unwrap();
+    let mut proxy_storage: SgxMutexGuard<PolkadexAccountsStorage> = mutex
+        .lock()
+        .map_err(|_| AccountRegistryError::CouldNotGetMutex)?;
     Ok(proxy_storage.remove_proxy(main_acc, proxy))
 }
 
-pub fn load_proxy_registry() -> Result<&'static SgxMutex<PolkadexAccountsStorage>, GatewayError> {
+pub fn load_proxy_registry(
+) -> Result<&'static SgxMutex<PolkadexAccountsStorage>, AccountRegistryError> {
     let ptr =
         GLOBAL_ACCOUNTS_STORAGE.load(Ordering::SeqCst) as *mut SgxMutex<PolkadexAccountsStorage>;
     if ptr.is_null() {
-        return Err(GatewayError::UnableToLoadPointer);
+        error!("Null pointer to polkadex account registry");
+        return Err(AccountRegistryError::CouldNotLoadRegistry);
```

Comment:
added some error! messages for easier error location during testing openfinex calls